Touch Bar Simulator/Glue.swift

```diff
@@ -0,0 +1,84 @@
+import Foundation
+import Defaults
+
+extension Defaults {
+	@discardableResult
+	func observe<T: Codable, Weak: AnyObject>(_ key: Key<T>, tiedToLifetimeOf weaklyHeldObject: Weak, options: NSKeyValueObservingOptions = [.initial, .new, .old], handler: @escaping (KeyChange<T>) -> Void) -> DefaultsObservation {
```

Comment:
Was thinking recently that maybe we could make this more composable by removing the `tiedToLifetimeOf` arguments and instead make it a method on an extension of `DefaultsObservation`?

Something like this:

```swift
defaults.observe(.foo) { change in
	print(change)
}.tiedToLifetimeOf(self)
```